From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Fri, 23 Sep 2022 01:20:54 +0200
Subject: Do not strip debug info for CMAKE_BUILD_TYPE=RelWithDebInfo

If `CMAKE_BUILD_TYPE` is set to `RelWithDebInfo` the debug information
should not be stripped. Currently `strip -x` is called when
`RelWithDebInfo` is set. A useless `echo` is called when `Debug` is set.

Skip `strip -x` when `CMAKE_BUILD_TYPE` is set to `Debug` or
`RelWithDebInfo`.

Forwarded: https://github.com/audacity/audacity/pull/3672
Signed-off-by: Benjamin Drung <benjamin.drung@canonical.com>
---
 cmake-proxies/cmake-modules/AudacityFunctions.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake-proxies/cmake-modules/AudacityFunctions.cmake b/cmake-proxies/cmake-modules/AudacityFunctions.cmake
index a8f144f..6234fc7 100644
--- a/cmake-proxies/cmake-modules/AudacityFunctions.cmake
+++ b/cmake-proxies/cmake-modules/AudacityFunctions.cmake
@@ -482,11 +482,11 @@ function( audacity_module_fn NAME SOURCES IMPORT_TARGETS
    target_link_options( ${TARGET} PRIVATE ${LOPTS} )
    target_link_libraries( ${TARGET} PUBLIC ${LIBRARIES} )
 
-   if( NOT CMAKE_SYSTEM_NAME MATCHES "Windows" )
+   if( NOT CMAKE_SYSTEM_NAME MATCHES "Windows" AND NOT CMAKE_BUILD_TYPE MATCHES "Debug|RelWithDebInfo" )
       add_custom_command(
          TARGET "${TARGET}"
          POST_BUILD
-         COMMAND $<IF:$<CONFIG:Debug>,echo,strip> -x $<TARGET_FILE:${TARGET}>
+         COMMAND strip -x $<TARGET_FILE:${TARGET}>
       )
    endif()
 
